File: python/URI/1021.py
valor = float(input())
notas = int(valor)
cents = (valor-notas)*100
cents = int(cents)

n100 = notas/100
resto= notas%100
n50=resto/50
resto%=50
n20=resto/20
resto%=20
n10=resto/10
resto%=10
n5=resto/5
resto%=5
n2=resto/2
resto%=2

cents+=resto*100
m100=cents/100
resto=cents%100
m50=resto/50
resto%=50
m25=resto/25
resto%=25
m10=resto/10
resto%=10
m5=resto/5
resto%=5
m1=resto
print("NOTAS:")
print("%i nota(s) de R$ 100.00" % n100)
print("%i nota(s) de R$ 50.00" % n50)
print("%i nota(s) de R$ 20.00" % n20)
print("%i nota(s) de R$ 10.00" % n10)
print("%i nota(s) de R$ 5.00" % n5)
print("%i nota(s) de R$ 2.00" % n2)
print("MOEDAS:")
print("%i moeda(s) de R$ 1.00" % m100)
print("%i moeda(s) de R$ 0.50" % m50)
print("%i moeda(s) de R$ 0.25" % m25)
print("%i moeda(s) de R$ 0.10" % m10)
print("%i moeda(s) de R$ 0.05" % m5)
print("%i moeda(s) de R$ 0.01" % m1)

File: C/URI/1021.c
#include <stdio.h>
void main (){
	double valor;
	int n100,n50,n20,n10,n5,n2;
	int m100,m50,m25,m10,m5,m1;
	int real,cent;
	scanf("%lf",&valor);
	// conversão em inteiro
	real=(int)valor;
	cent=((valor-real)*100);
	
	// notas
	n100 = real/100;
	n50 = (real%100)/50;
	n20 = (((real%100)%50)/20);
	n10 = ((((real%100)%50)%20)/10);
	n5 = (((((real%100)%50)%20)%10)/5);
	n2 = ((((((real%100)%50)%20)%10)%5)/2);
	
	// moedas
	m100 = (((((((real%100)%50)%20)%10)%5)%2)/1);
	m50 = ((cent%100)/50);
	m25 = (((cent%100)%50)/25);
	m10 = ((((cent%100)%50)%25)/10);
	m5 = (((((cent%100)%50)%25)%10)/5);
	m1 = ((((((cent%100)%50)%25)%10)%5)/1);
	
	// print
	printf("NOTAS:\n");
	printf("%i nota(s) de R$ 100.00\n",n100);
	printf("%i nota(s) de R$ 50.00\n",n50);
	printf("%i nota(s) de R$ 20.00\n",n20);
	printf("%i nota(s) de R$ 10.00\n",n10);
	printf("%i nota(s) de R$ 5.00\n",n5);
	printf("%i nota(s) de R$ 2.00\n",n2);	
	printf("MOEDAS:\n");
	printf("%i moeda(s) de R$ 1.00\n",m100);
	printf("%i moeda(s) de R$ 0.50\n",m50);
	printf("%i moeda(s) de R$ 0.25\n",m25);
	printf("%i moeda(s) de R$ 0.10\n",m10);
	printf("%i moeda(s) de R$ 0.05\n",m5);
	printf("%i moeda(s) de R$ 0.01\n",m1);
	
}